README.md

```diff
@@ -1,2 +1,191 @@
-# terraform-google-kubernetes-engine
-A Cloud Foundation Toolkit Module: Opinionated Google Cloud Platform project creation and configuration with Shared VPC, IAM, APIs, etc.
+# Terraform Kubernetes Engine Module
+
+This module handles opinionated Google Cloud Platform Kubernetes Engine cluster creation and configuration with Node Pools, IP MASQ, Network Policy, etc.
+
+## Requirements
+### Kubectl
+- [kubectl](https://github.com/kubernetes/kubernetes/releases) 1.9.x
+### Terraform plugins
+- [Terraform](https://www.terraform.io/downloads.html) 0.10.x
+- [terraform-provider-google](https://github.com/terraform-providers/terraform-provider-google) plugin v1.8.0
+
+### Configure a Service Account
+In order to execute this module you must have a Service Account with the following:
+
+#### Roles
+The service account with the following roles:
+- roles/compute.viewer on the project
+- roles/container.clusterAdmin on the project
+
+### Enable API's
+In order to operate with the Service Account you must activate the following APIs on the project where the Service Account was created:
+
+- Compute Engine API - compute.googleapis.com
+- Kubernetes Engine API - container.googleapis.com
+
+## Install
+
+### Terraform
+Be sure you have the correct Terraform version (0.10.x), you can choose the binary here:
+- https://releases.hashicorp.com/terraform/
+
+## Usage
+There are multiple examples included in the [examples](./examples/) folder but simple usage is as follows:
+
+```hcl
+module "gke" {
+  source                     = "github.com/terraform-google-modules/terraform-google-kubernetes-engine"
+  project_id                 = "<PROJECT ID>"
+  name                       = "gke-test-1"
+  region                     = "us-central1"
+  network                    = "vpc-01"
+  subnetwork                 = "us-central1-01"
+  ip_range_pods              = "us-central1-01-gke-01-pods"
+  ip_range_services          = "us-central1-01-gke-01-services"
+  node_service_account       = "project-service-account@<PROJECT ID>.iam.gserviceaccount.com"
```

Comment:
@ryanckoch Still need to change this to apply it on a node_pool level.

Reply:
Now implemented.

Reply:
Instead of requiring that you specify this, we should use the service account from the first node_pool.

Reply:
It's currently set as an optional variable so you could set one service account for all pools or set at the node pool level (or none at all). If you don't agree with that logic can definitely remove it.

Reply:
I'd prefer to have it only at the node-pool level, no need to specify a default for all.

Reply:
Removed the logic.